src/main/java/com/devonfw/demoquarkus/logic/UcFindAnimalImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.devonfw.demoquarkus.logic;
+
+import com.devonfw.demoquarkus.domain.model.AnimalEntity;
+import com.devonfw.demoquarkus.domain.repo.AnimalRepository;
+import com.devonfw.demoquarkus.service.v1.mapper.AnimalMapper;
+import com.devonfw.demoquarkus.service.v1.model.AnimalDto;
+import com.devonfw.demoquarkus.service.v1.model.AnimalSearchCriteriaDto;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+
+@Named
+@Transactional
+@Slf4j
+public class UcFindAnimalImpl implements UcFindAnimal {
+    @Inject
+    AnimalRepository animalRepository;
+
+    @Inject
+    AnimalMapper mapper;
+
+    @Override
+    public Page<AnimalDto> findAnimals(AnimalSearchCriteriaDto dto) {
+        Iterable<AnimalEntity> animalsIterator = this.animalRepository.findAll();
+        List<AnimalEntity> animals = new ArrayList<AnimalEntity>();
+        animalsIterator.forEach(animals::add);
+        List<AnimalDto> animalsDto = this.mapper.map(animals);
+        return new PageImpl<>(animalsDto, PageRequest.of(dto.getPageNumber(), dto.getPageSize()), animalsDto.size());
+    }
+
+    @Override
+    public Page<AnimalDto> findAnimalsByCriteriaApi(AnimalSearchCriteriaDto dto) {
+        List<AnimalEntity> animals = this.animalRepository.findAllCriteriaApi(dto).getContent();
+        List<AnimalDto> animalsDto = this.mapper.map(animals);
+        return new PageImpl<>(animalsDto, PageRequest.of(dto.getPageNumber(), dto.getPageSize()), animalsDto.size());
+    }
+
+    @Override
+    public Page<AnimalDto> findAnimalsByQueryDsl(AnimalSearchCriteriaDto dto) {
+        List<AnimalEntity> animals = this.animalRepository.findAllQueryDsl(dto).getContent();
+        List<AnimalDto> animalsDto = this.mapper.map(animals);
+        return new PageImpl<>(animalsDto, PageRequest.of(dto.getPageNumber(), dto.getPageSize()), animalsDto.size());
+    }
+
+    @Override
+    public Page<AnimalDto> findAnimalsByNameQuery(AnimalSearchCriteriaDto dto) {
+        List<AnimalEntity> animals = this.animalRepository.findByNameQuery(dto).getContent();
```

Comment:
bug pattern also here:
https://github.com/devonfw-sample/devon4quarkus-reference/blob/master/src/main/java/com/devonfw/demoquarkus/domain/repo/AnimalFragmentImpl.java#L81
Has to be fixed to properly support paging.